tests/clients/settings/test_incydr_device_settings.py

```diff
@@ -0,0 +1,106 @@
+import pytest
+from tests.clients.conftest import param
+
+from py42.clients.settings import get_val
+from py42.clients.settings.device_settings import IncydrDeviceSettings
+
+TEST_USER_ID = 13548744
+TEST_COMPUTER_ID = 4290210
+TEST_COMPUTER_GUID = "42000000"
+TEST_COMPUTER_ORG_ID = 424242
+TEST_DEVICE_VERSION = 4200000000001
+TEST_COMPUTER_NAME = "Incydr Settings Test Device"
+
+INCYDR_DEVICE_DICT_W_SETTINGS = {
+    "computerId": 4290210,
+    "name": "Incydr Settings Test Device",
+    "osHostname": "DESKTOP-I2SKEML",
+    "guid": "42000000",
+    "type": "COMPUTER",
+    "status": "Active",
+    "active": True,
+    "blocked": False,
+    "alertState": 0,
+    "alertStates": ["OK"],
+    "userId": 13548744,
+    "userUid": "1010090007721726158",
+    "orgId": 424242,
+    "orgUid": "985187827481212202",
+    "computerExtRef": None,
+    "notes": None,
+    "parentComputerId": None,
+    "parentComputerGuid": None,
+    "lastConnected": "2021-07-22T12:38:54.854Z",
+    "osName": "Windows",
+    "osVersion": "10.0.19041",
+    "osArch": "x86-64",
+    "address": "192.168.223.128",
+    "remoteAddress": "50.237.14.12",
+    "javaVersion": "",
+    "modelInfo": "",
+    "timeZone": "(UTC-06:00) Central Time (US & Cana",
+    "version": 4200000000001,
+    "productVersion": "0.0.1",
+    "buildVersion": 3531,
+    "creationDate": "2021-06-04T14:30:51.392Z",
+    "modificationDate": "2021-07-22T12:38:54.854Z",
+    "loginDate": "2021-06-04T14:30:51.362Z",
+    "service": "Artemis",
+    "orgSettings": {"securityKeyLocked": False},
+}
+
+
+class TestIncydrDeviceSettings:
+    device_settings = IncydrDeviceSettings(INCYDR_DEVICE_DICT_W_SETTINGS)
+
+    @pytest.mark.parametrize(
+        "param",
+        [
+            ("name", TEST_COMPUTER_NAME),
+            ("computer_id", TEST_COMPUTER_ID),
+            ("device_id", TEST_COMPUTER_ID),
+            ("guid", TEST_COMPUTER_GUID),
+            ("user_id", TEST_USER_ID),
+            ("org_id", TEST_COMPUTER_ORG_ID),
+            ("version", TEST_DEVICE_VERSION),
+        ],
+    )
+    def test_device_settings_properties_return_expected_value_and_cannot_be_changed(
+        self, param
+    ):
+        name, expected_value = param
+        assert getattr(self.device_settings, name) == expected_value
+        with pytest.raises(AttributeError):
+            setattr(self.device_settings, name, expected_value)
+
+    @pytest.mark.parametrize("param", [("notes", None), ("external_reference", None)])
```

Comment:
We should add examples with real test strings here too, as we could hypothetically change the implementation to something like:

```
@property
    def notes(self):
        pass
```
and this test would still pass.